domenetjenester/dokumentbestiller/src/main/java/no/nav/foreldrepenger/dokumentbestiller/infobrev/FeilPraksisUtsettelseRepository.java

```diff
@@ -0,0 +1,211 @@
+package no.nav.foreldrepenger.dokumentbestiller.infobrev;
+
+import java.util.List;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+
+import org.hibernate.jpa.HibernateHints;
+
+import no.nav.foreldrepenger.behandlingslager.behandling.BehandlingÅrsakType;
+
+/**
+ * Spesialmetoder for å hente opp saker og personer som er kandidat for å sende
+ * informasjonsbrev om feil praksis utsettelse og gradering
+ */
+
+@ApplicationScoped
+public class FeilPraksisUtsettelseRepository {
+
+    private EntityManager entityManager;
+
+    FeilPraksisUtsettelseRepository() {
+        // for CDI proxy
+    }
+
+    @Inject
+    public FeilPraksisUtsettelseRepository(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    private static final String QUERY_MOR = """
+        select * from (
+        select f.id fid
+        from fagsak f
+           join behandling b on fagsak_id = f.id
+           join behandling_resultat br  on b.id = br.behandling_id
+           join behandling_vedtak bv on bv.behandling_resultat_id = br.id
+           join (select beh.fagsak_id fsmax, max(bvsq.opprettet_tid) maxbr from behandling beh
+                join behandling_resultat brsq on brsq.behandling_id=beh.id
+                join behandling_vedtak bvsq on bvsq.behandling_resultat_id = brsq.id
+                where beh.behandling_type in ('BT-002','BT-004') and beh.behandling_status in ('AVSLU', 'IVED')
+                group by beh.fagsak_id) on (fsmax=b.fagsak_id and bv.opprettet_tid = maxbr)
+           join uttak_resultat ur on (ur.behandling_resultat_id = br.id and ur.aktiv='J')
+        where 1=1
+           and b.behandling_type in ('BT-002','BT-004') and b.behandling_status in ('AVSLU', 'IVED')
+           and nvl(ur.overstyrt_perioder_id, ur.opprinnelig_perioder_id) in (
+              select uttak_resultat_perioder_id from uttak_resultat_periode peri
+              where (peri.gradering_avslag_AARSAK = '4501' and samtidig_uttaksprosent is null
+                     and exists (select * from uttak_resultat_periode_akt peria
+                          where peria.uttak_resultat_periode_id = peri.id and utbetalingsprosent < 100)
+              ) or (peri.PERIODE_RESULTAT_AARSAK in ( '4005','4081','4082')
```

Comment:
Samme årsaker sendes opp til fpoversikt/selvbetjening.
Men vi har ikke filtrering på 
`and b.behandling_type in ('BT-002','BT-004') and b.behandling_status in ('AVSLU', 'IVED')`
som jeg ser samtlige queries har her.

Noe filtrering som må legges inn for innsyn?

Reply:
Fpoversikt bryr seg bare om vedtak/åpen behandling av type bt-002 og bt-004, så vi trenger ikke å gjøre noe der